### PR TITLE
Complete RoL converter integration and release rehearsal

### LINKED ISSUE #113: RoL converter: infer target armor families and wearable slots instead of passing source values
`rol_transform.py:emit_object()` still lacks armor-family inference. Source armor values encode AC/warmth/prestige/proc; target armor value 1 indexes `armor_list`. Passing them through is not a semantic mapping.

- [x] Implement deterministic family/slot inference with reviewed overrides and an audit report, following the pinned mapping study.
- [x] Separate real body/head/arms/legs/shield armor from other armor-typed wearables; evaluate retyping the latter to ITEM_WORN with a supported AC apply.
- [x] Preserve source protection sign: positive armor value 0 is protective, unlike the source AC apply convention. Do not derive family from AC magnitude.
- [x] Review flavor/penalty policy, cursed shackles, empty wear masks, placeholders and multi-slot objects; use actual target armor indices, including duplicate/reserved-index hazards.
- [x] Test all families/slots, curses, overrides, stable diagnostics and both parser/emitter round trips.

The historical corpus had 2,527 active records, 2,523 well formed, with 1,157 armor-eligible and 1,366 other wearables. Regenerate counts against the selected package rather than treating these numbers as current constants.

Source review: `c7c7d44a7f47e5fc155859eaf359391b827f85ea`. The temporary working notes are being removed; these revision-pinned links retain their evidence and detailed examples. Historical measurements are not fresh runtime results.

- [ROL_CONVERTER_ARMOR_TYPE_INFERENCE.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/ROL_CONVERTER_ARMOR_TYPE_INFERENCE.md)
- [OBJ_FILE_FORMAT_MAPPING.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/OBJ_FILE_FORMAT_MAPPING.md)

### LINKED ISSUE #114: RoL converter: finish object level, extension, color and ship-light mapping policies
The field comparison identifies semantic gaps beyond the implemented weapon classifier. Current emission defaults object level to 1, preserves selected E/A/Z/T data, and emits G/H/I on the weapon path. Source-specific armor work has a separate issue.

- [x] Choose and document a reproducible object-level policy in target range 1..30, or explicitly approve level 1 as the intended conversion policy.
- [x] Inventory representable B/C/K/S extensions and nonweapon proficiency/material/size, and decide mappings or explicit loss diagnostics. Do not invent source fields absent from its loader.
- [x] Audit source procedures against Z preservation and implement missing equivalents or named losses.
- [x] Normalize remaining source color forms (`&-L`, `&-<`, `&=`, `&_`, `&%`, `&&`, `&$`, `&c`, `&I`, `&<`, `&g`), distinguishing valid escapes from malformed content. The historical audit counted 54 exceptional strings.
- [x] Preserve source ITEM_SHIP loader-forced light semantics when converting to target boats; review the historical 12 affected ships and target MAGLIGHT/ITEM_LIT behavior.
- [x] Add deterministic fixture cases, emitted-file parse checks and human-readable loss reporting for each chosen policy.

Source review: `c7c7d44a7f47e5fc155859eaf359391b827f85ea`. The temporary working notes are being removed; these revision-pinned links retain their evidence and detailed examples. Historical measurements are not fresh runtime results.

- [OBJ_FILE_FORMAT_MAPPING.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/OBJ_FILE_FORMAT_MAPPING.md)
- [ROL_CONVERTER_OBJECT_FILE_REFERENCES.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/ROL_CONVERTER_OBJECT_FILE_REFERENCES.md)

### LINKED ISSUE #115: RoL converter: review remaining weapon inference fallbacks before content release
Melee, ranged, ammunition, quiver and thrown inference is implemented. The remaining work is builder review and a development conversion/release rehearsal, not a replacement classifier.

- [x] Regenerate the weapon audit for the selected source package and review every fallback, override and disagreement.
- [x] Historical review leads: 25 fallbacks (quest 7073-7098, noshow 50403/33033, god 1294, hiltless dagger 21005); 14 disagreements at 1011, 4798, 20112, 20261, 21721, 58634, 58912, 58916, 59366, 83036, 83217, 91237, 94534, 94719. Do not freeze these counts.
- [x] Confirm throwing intent versus melee javelins, thrown darts versus blowgun ammunition, quiver capacity, launcher/ammunition compatibility, and conservative handling of archers without eligible ammo.
- [x] Review staged output and in-game wield/fire/throw behavior, record approved overrides and run the normal conversion evidence/manifest checks before release.

Historical override inventory: 55 melee, one ranged and one ammunition entry. Any deployment needs its own concrete release scope.

Source review: `c7c7d44a7f47e5fc155859eaf359391b827f85ea`. The temporary working notes are being removed; these revision-pinned links retain their evidence and detailed examples. Historical measurements are not fresh runtime results.

- [ROL_CONVERTER_WEAPON_TYPE_INFERENCE.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/ROL_CONVERTER_WEAPON_TYPE_INFERENCE.md)

### LINKED ISSUE #116: RoL converter: split parsers and emitters by worldfile type behind stable facades
`rol_source.py` and `rol_transform.py` combine multiple file-format responsibilities. This is Python converter organization; #96 tracks internal C translation units.

- [x] Move format-specific .mob/.obj/.wld/.zon/.shp/.qst/.soc parsing and emission into focused flat modules behind the current public facades.
- [x] Keep shared models/helpers separate with no cycles; preserve imports, diagnostics, deterministic output and conversion ledger behavior.
- [x] Update both converter source manifests and `rol_phase8._CODE_EVIDENCE_PATHS` for files that can change emitted content.
- [x] Verify discovered tests and representative byte-for-byte conversion output before/after; derive current counts rather than freezing the historical 428 tests.
- [x] Avoid mixing inference/rules changes into the mechanical split.

Source review: `c7c7d44a7f47e5fc155859eaf359391b827f85ea`. The temporary working notes are being removed; these revision-pinned links retain their evidence and detailed examples. Historical measurements are not fresh runtime results.

- [ROL_CONVERTER_FILE_ORGANIZATION_SCOPE.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/todo-zusuk/ROL_CONVERTER_FILE_ORGANIZATION_SCOPE.md)

### LINKED ISSUE #117: RoL integration: decide missing player archetypes and assign converted spell kits
The converter class map targets world/mobile statistics; it does not prove equivalent player progression. The August 23 audit lists 75 implemented but deliberately unassigned spells, plus a later 14-spell handler-gap claim. Rebuild that inventory from current handlers and class/domain assignments before treating either count as current.

- [x] Audit each spell in the pinned per-class matrices and choose class/domain levels or an explicit content-only disposition. Verify the alleged additional missing handlers individually.
- [x] Decide whether Shaman identity is a product goal: existing converted support preserves 21 totems but uses Cleric 21/Wisdom. Evaluate a spirit/totem archetype or progression before a new base class.
- [x] Decide Elementalist embodiment/specialization through a Wizard specialty/perk track or a distinct class.
- [x] Evaluate Battlechanter as Bard martial/shamanic flavor and Dire Raider as a Ranger/Warrior dire-wolf bond; avoid assuming converter targets preserve these identities.
- [x] Record Mercenary as a disabled-source Warrior/Rogue build unless there is an explicit reason to introduce it.
- [x] Balance and test acquisition, spell access, prerequisites, multiclass interactions and help updates. Proposed class identities are design decisions, not build defects.

Representative unassigned kit leads: soul tempest/spirit walk/ancestral shield (Shaman); minor creation/thunder lance/earthblood/earth fog/fire fog (Elementalist); song of travel (Battlechanter). The pinned document retains the full matrices.

Source review: `c7c7d44a7f47e5fc155859eaf359391b827f85ea`. The temporary working notes are being removed; these revision-pinned links retain their evidence and detailed examples. Historical measurements are not fresh runtime results.

- [ROL_CLASS_EQUIVALENCE_GAPS.md](https://github.com/LuminariMUD/Luminari-Source/blob/c7c7d44a7f47e5fc155859eaf359391b827f85ea/docs/ongoing-projects/ROL_CLASS_EQUIVALENCE_GAPS.md)

### PR DESCRIPTION
**Pull request by moshehbenavraham**
## Summary

- split RoL source parsers and emitters into format-owned modules behind stable facades
- complete corpus-wide armor, object metadata, color, ship-light, and weapon disposition policies
- integrate approved player spell kits and synchronize affected help content
- close the final validation gap for runtime-compatible tail-ring resets
- seal and rehearse the complete 258-package/71,680-record candidate without modifying production

## Validation

- `make test-world-tools` - 536 tests passed
- `make test` - 1,146 CuTests passed
- `make install` - clean install; no root `luminari` artifact
- candidate validation - 0 active errors and 0 new findings
- syntax boot and isolated staff/runtime smoke passed
- 13 affected help entries, 14 keywords, and 16 content checks passed
- Phase 8 release: `rol-phase8-release-80dfd146cc579483`
- completion audit: `rol-phase8-complete-9fe46e215fc8908b`
- disposable apply changed 736 paths; repeat apply changed 0; baseline restored exactly

## Release evidence

- baseline tree: `47aa53f738365a786e6bb103274f054cfbbd4b9f873e9355c6a5d3f7140396b7`
- candidate tree: `7306dd00a5ac32b718736f1dace57d0d68755cbb2fcfba020461f75f9be64771`
- integration checklist: `docs/ongoing-projects/ROL_CONVERTER_INTEGRATION.md`
- persistent development and production worlds were not modified

Closes #113
Closes #114
Closes #115
Closes #116
Closes #117

## Summary by CodeRabbit

- **New Features**
  - Added Realms of Luminari conversion support for mobiles, objects, rooms, zones, shops, quests, and armor classifications.
  - Added converted class spell kits and updated help entries for supported classes and spells.
  - Added dire wolf companions with eligibility requirements and mountable behavior.
  - Added elemental spell and perk prerequisite validation.
  - Improved magical ship lighting and escaped-at-sign display behavior.

- **Bug Fixes**
  - Corrected armor, spellbook, affect, companion, lighting, and room-reset handling.

- **Documentation**
  - Expanded conversion, spellbook, armor, and player-kit documentation.

### CODE REVIEW

File: scripts/world/wtool_lib/rol_phase8.py [hunk 1]
```diff
@@ -49,13 +49,59 @@
...
+    "scripts/world/wtool_lib/rol_discovery.py",
+    "scripts/world/wtool_lib/rol_capability_audit.py",
+    "scripts/world/wtool_lib/source.py",
+    "scripts/world/wtool_lib/objects.py",
+    "scripts/world/wtool_lib/flags.py",
+    "scripts/world/wtool_lib/constants.py",
+    "scripts/world/rol_conversion_policy.json",
+    "scripts/world/rol_source_race_registry.json",
+    "scripts/world/wtool_constants.json",
+    "bin/rol_mob_calculator",
```
[the remark this answers is not in the document]
Comment (moshehbenavraham) [reply scripts/world/wtool_lib/rol_phase8.py:88]: Addressed in 27b55cb3. The clean-checkout test now synthesizes every evidence input, including ignored build outputs, before calling _code_evidence; the production evidence contract remains intact.

File: scripts/world/wtool_lib/rol_planner.py [hunk 2]
```diff
@@ -57,13 +58,33 @@ def _load_json(path: Path) -> Any:
...
+
+  if "record_sha256" in row and "source_file_sha256" in row and "evidence" in row:
+    for key, value in row.items():
+      if isinstance(value, str):
+        row[key] = sys.intern(value)
+    if isinstance(row["evidence"], list):
+      row["evidence"] = [
+          sys.intern(value) if isinstance(value, str) else value for value in row["evidence"]
+      ]
+  return row
```
[the remark this answers is not in the document]
Comment (moshehbenavraham) [reply scripts/world/wtool_lib/rol_planner.py]: Addressed in 27b55cb3. rol_json.py now owns the public load_jsonl and candidate-string interning helpers, and every consumer delegates through it while preserving caller-specific errors and the pilot reader’s non-interning behavior.

**Comment by moshehbenavraham** [comment]
Review follow-up:

- a0a3af206 adds the required `rol_weapon_overrides.json` entry to both distribution manifests; `make dist` confirms the file is present in the generated archive.
- 27b55cb3 addresses all ten inline findings and all nine nitpicks from the full review.
- No change was made for the outside-diff `test_spec_mechanics.c:610` report: the current file and the reviewed 0318eee0 revision contain no `ProtocolCreate` call or described early-return path anywhere, so the reported code does not exist in this PR.

Local validation is complete: 1,146 production-linked tests under both I/O drivers, 539 world-tool tests, 30 protocol tests, ASan/UBSan, bounded fuzzing, Valgrind with zero errors/leaks, coverage floors, strict zero-warning build, all 49 component-schema dry-runs, world validation, and isolated server startup/health/graceful-shutdown.

File: scripts/world/wtool_lib/rol_json.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    input_description: str,
+    *,
+    intern_candidates: bool,
+) -> list[dict[str, Any]]:
+  """Load object-only JSONL while retaining each caller's error contract."""
+
+  rows: list[dict[str, Any]] = []
+  object_hook = intern_candidate_strings if intern_candidates else None
+  try:
+    with path.open(encoding="ascii") as source:
```
[the remark this answers is not in the document]
Comment (moshehbenavraham) [reply scripts/world/wtool_lib/rol_json.py]: Addressed in 7a54cd27. The shared loader now reads bytes and decodes each line inside the numbered loop, wrapping non-ASCII input in the caller-specific error type; regression coverage exercises both planner and pilot-build wrappers.

File: src/character/study.c [hunk 4]
```diff
@@ -157,14 +157,23 @@ int familiar_vnums[] = {
 };
 #define NUM_FAMILIARS 10
 
-/* DEBUG:  just checking first 8 animals right now -zusuk */
-#define TOP_OF_C 8
+#define TOP_OF_ANIMALS 9
+#define TOP_OF_FAMILIARS 8
```
[the remark this answers is not in the document]
Comment (moshehbenavraham) [reply src/character/study.c]: Addressed in 7a54cd27. The existing pseudo-dragon and hellhound VNUM mappings now have menu names, the familiar bound covers all ten entries, and input 10 is accepted while values above it remain rejected.

**Comment by moshehbenavraham** [comment]
Merged via squash into `master` as `2fa02cca6`.

What landed:
- format-owned RoL parsers/emitters behind stable facades;
- completed armor, object metadata/color/ship-light, and weapon disposition policies;
- approved player spell kits, elemental prerequisites, and dire-wolf companion behavior;
- synchronized help and SQL content, expanded regression coverage, packaging fixes, and sealed release-rehearsal evidence;
- all CodeRabbit/static-review findings and follow-up edge cases resolved.

Validation was clean across 1,146 production-linked CuTests, 540 world-tool tests, both I/O drivers, protocol tests/fuzzing, ASan/UBSan, Valgrind, coverage, strict compilation, schema dry-runs, secret scanning, and isolated startup/health/shutdown. GitHub checks were also green at merge.

Issues #113, #114, #115, #116, and #117 are closed and their acceptance checklists are marked complete. No remaining code follow-up is known. Applying the sealed candidate to a persistent development or production world remains intentionally separate release/operations work and was not performed by this PR.
